From 5b1e7fe6bb954ab952b64775232985d0c780cb3f Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jiri@resnulli.us>
Date: Tue, 1 Jan 2013 03:30:15 +0000
Subject: [PATCH 466/858] net: set dev->addr_assign_type correctly

Not a bitfield, but a plain value.

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e41b2d7fe7803e85e1202d0eb172717d7bf1bbaf)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/atheros/atl1c/atl1c_main.c | 2 +-
 drivers/net/ethernet/atheros/atlx/atl1.c        | 2 +-
 drivers/net/ethernet/ethoc.c                    | 2 +-
 drivers/net/ethernet/lantiq_etop.c              | 2 +-
 include/linux/etherdevice.h                     | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/atheros/atl1c/atl1c_main.c b/drivers/net/ethernet/atheros/atl1c/atl1c_main.c
index f06d32d..d1c2f54 100644
--- a/drivers/net/ethernet/atheros/atl1c/atl1c_main.c
+++ b/drivers/net/ethernet/atheros/atl1c/atl1c_main.c
@@ -2591,7 +2591,7 @@ static int atl1c_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  }
  if (atl1c_read_mac_addr(&adapter->hw)) {
   /* got a random MAC address, set NET_ADDR_RANDOM to netdev */
-  netdev->addr_assign_type |= NET_ADDR_RANDOM;
+  netdev->addr_assign_type = NET_ADDR_RANDOM;
  }
  memcpy(netdev->dev_addr, adapter->hw.mac_addr, netdev->addr_len);
  if (netif_msg_probe(adapter))
diff --git a/drivers/net/ethernet/atheros/atlx/atl1.c b/drivers/net/ethernet/atheros/atlx/atl1.c
index 71b3d7d..5b0d993 100644
--- a/drivers/net/ethernet/atheros/atlx/atl1.c
+++ b/drivers/net/ethernet/atheros/atlx/atl1.c
@@ -3053,7 +3053,7 @@ static int atl1_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  /* copy the MAC address out of the EEPROM */
  if (atl1_read_mac_addr(&adapter->hw)) {
   /* mark random mac */
-  netdev->addr_assign_type |= NET_ADDR_RANDOM;
+  netdev->addr_assign_type = NET_ADDR_RANDOM;
  }
  memcpy(netdev->dev_addr, adapter->hw.mac_addr, netdev->addr_len);
 
diff --git a/drivers/net/ethernet/ethoc.c b/drivers/net/ethernet/ethoc.c
index 8db1c06..f380bb7 100644
--- a/drivers/net/ethernet/ethoc.c
+++ b/drivers/net/ethernet/ethoc.c
@@ -1068,7 +1068,7 @@ static int ethoc_probe(struct platform_device *pdev)
  }
 
  if (random_mac)
-  netdev->addr_assign_type |= NET_ADDR_RANDOM;
+  netdev->addr_assign_type = NET_ADDR_RANDOM;
 
  /* register MII bus */
  priv->mdio = mdiobus_alloc();
diff --git a/drivers/net/ethernet/lantiq_etop.c b/drivers/net/ethernet/lantiq_etop.c
index c124e67..cd3d2c0 100644
--- a/drivers/net/ethernet/lantiq_etop.c
+++ b/drivers/net/ethernet/lantiq_etop.c
@@ -655,7 +655,7 @@ ltq_etop_init(struct net_device *dev)
 
  /* Set addr_assign_type here, ltq_etop_set_mac_address would reset it. */
  if (random_mac)
-  dev->addr_assign_type |= NET_ADDR_RANDOM;
+  dev->addr_assign_type = NET_ADDR_RANDOM;
 
  ltq_etop_set_multicast_list(dev);
  err = ltq_etop_mdio_init(dev);
diff --git a/include/linux/etherdevice.h b/include/linux/etherdevice.h
index 243eea1..1a43e1b 100644
--- a/include/linux/etherdevice.h
+++ b/include/linux/etherdevice.h
@@ -192,7 +192,7 @@ static inline void eth_zero_addr(u8 *addr)
  */
 static inline void eth_hw_addr_random(struct net_device *dev)
 {
- dev->addr_assign_type |= NET_ADDR_RANDOM;
+ dev->addr_assign_type = NET_ADDR_RANDOM;
  eth_random_addr(dev->dev_addr);
 }
 
-- 
1.8.3.1